server/service/src/processors/transfer/invoice/test.rs

```diff
@@ -1269,24 +1317,41 @@ fn check_line(connection: &StorageConnection, inbound_id: &str, outbound_line: &
     assert_eq!(inbound_line.pack_size, outbound_line.pack_size);
     assert_eq!(inbound_line.number_of_packs, outbound_line.number_of_packs);
     assert_eq!(inbound_line.note, outbound_line.note);
-    assert_eq!(inbound_line.r#type, InvoiceLineType::StockIn);
+
+    match outbound_line.r#type {
+        InvoiceLineType::Service => {
+            assert_eq!(inbound_line.r#type, InvoiceLineType::Service);
+            assert_eq!(
+                inbound_line.total_before_tax,
+                outbound_line.total_before_tax
+            );
+            assert_approx_eq::assert_approx_eq!(
```

Comment:
Test was failing will `110.0 != 110.00000000000001` ... in DB the value looks correct... is assert_approx ok here or should I have some rounding in the transfer logic?

Reply:
I think approx is fine - trying to approximate floating point numbers in binary 🤔 
If we change, I'd update `calculate_total_after_tax` and have that round the numbers slightly. But I'd prefer not to - just have dodgy floats in the database and round in the UI layer.

Though I would also do this:

```
total_after_tax: calculate_total_after_tax(total_before_tax, tax_percentage),
```

in `/server/service/src/processors/transfer/invoice/common.rs:65` to make the calculation consistent.